### PR TITLE
revise function for selecting credit name

### PR DESCRIPTION
**Pull request by 70ray**
This addresses issue #1231.
Select elements by id rather than from form.
Change credit name input to hidden rather than disabled if not applicable.
If this works as it appears to be intended the code could be simplified by removing comments which no longer apply  and revising the function `dropdown_select_values_and_labels()` which appears to be used in only one place.

Sandbox at: https://www.pgdp.org/~rp31/c.branch/pref_credit

### CODE REVIEW

**Review by chrismiceli — APPROVED** [review]

**Comment by 70ray** [comment]
Sandbox updated.

File: userprefs.php [hunk 1]
```diff
@@ -951,54 +945,18 @@ function _show_credit_name_adhoc()
...
-function dropdown_select_values_and_labels($field_name, $current_value, $values, $labels, $on_change = '')
-{
-    global $event_id, $window_onload_event;
-
-    $function_name = 'event' . ++$event_id;
-    $jscode = "var f=document.forms[0];\nvar t=f.$field_name;\n$on_change";
+    $credit_options = [
+        'real_name' => _('Real Name'),
+        'username' => _('Username'),
+        'other' => _('Other:'),
```
Comment (cpeel) [inline userprefs.php:951]:
```suggestion
        'other' => _('Other') . ':',
```
We don't want the `:` inside the translated string.

File: userprefs.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+window.addEventListener("DOMContentLoaded", () => {
```
Comment (cpeel) [inline userprefs.js:1]: Let's move this to `scripts/` so it's not hanging out here in the root. `userprefs.php` probably shouldn't be here either but that ship has sailed.

**Review by cpeel — COMMENTED** [review]
This is _much_ better than what was there before!

**Comment by 70ray** [comment]
Sandbox updated.

File: scripts/userprefs.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+window.addEventListener("DOMContentLoaded", () => {
+
+    let creditNameSelector = document.getElementById('credit_name');
+    let creditOtherInput = document.getElementById('credit_other');
+
+    function maybeShowOther() {
+        creditOtherInput.hidden = (creditNameSelector.value != 'other');
```
Comment (chrismiceli) [inline scripts/userprefs.js:7]: *minor* unnecessary parentheses. Also I think eslint may complain or warn about not using `!==` over `!=`.

**Review by chrismiceli — APPROVED** [review]

**Review by srjfoo — APPROVED** [review]

File: scripts/userprefs.js (hunk 3 above)
Comment (70ray) [reply scripts/userprefs.js:7]: I think the parentheses make it more readable. probably better to always use !==.

**Review by chrismiceli — APPROVED** [review]

**Review by cpeel — APPROVED** [review]

**Pull request merged**